### ISSUE TITLE
mvn install on core-java-8 leads to errors

### ISSUE THREAD

**Issue by rolfschumacher** [eugenp/tutorials#6023]
switched to Java-8. Entered "mvn clean install -Dgib.enabled=false", resulted to 

[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 9.926 s
[INFO] Finished at: 2018-12-29T17:55:32+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.21.0:test (default-test) on project core-java-8: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/rsc/workspace/git/tutorials/core-java-8/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[ERROR] ExecutionException The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/rsc/workspace/git/tutorials/core-java-8 && /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -jar /home/rsc/workspace/git/tutorials/core-java-8/target/surefire/surefirebooter2097030882569557424.jar /home/rsc/workspace/git/tutorials/core-java-8/target/surefire 2018-12-29T17-55-30_971-jvmRun1 surefire1674629841317450003tmp surefire_12452190257193524138tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] ExecutionException The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/rsc/workspace/git/tutorials/core-java-8 && /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -jar /home/rsc/workspace/git/tutorials/core-java-8/target/surefire/surefirebooter7607318203199118946.jar /home/rsc/workspace/git/tutorials/core-java-8/target/surefire 2018-12-29T17-55-30_971-jvmRun3 surefire7901105064807722702tmp surefire_0262932308020382045tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] ExecutionException The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/rsc/workspace/git/tutorials/core-java-8 && /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -jar /home/rsc/workspace/git/tutorials/core-java-8/target/surefire/surefirebooter3119033343586266928.jar /home/rsc/workspace/git/tutorials/core-java-8/target/surefire 2018-12-29T17-55-30_971-jvmRun2 surefire5414877350063350396tmp surefire_24571033818249870520tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: ExecutionException The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/rsc/workspace/git/tutorials/core-java-8 && /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -jar /home/rsc/workspace/git/tutorials/core-java-8/target/surefire/surefirebooter2097030882569557424.jar /home/rsc/workspace/git/tutorials/core-java-8/target/surefire 2018-12-29T17-55-30_971-jvmRun1 surefire1674629841317450003tmp surefire_12452190257193524138tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] ExecutionException The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/rsc/workspace/git/tutorials/core-java-8 && /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -jar /home/rsc/workspace/git/tutorials/core-java-8/target/surefire/surefirebooter7607318203199118946.jar /home/rsc/workspace/git/tutorials/core-java-8/target/surefire 2018-12-29T17-55-30_971-jvmRun3 surefire7901105064807722702tmp surefire_0262932308020382045tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] ExecutionException The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/rsc/workspace/git/tutorials/core-java-8 && /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -jar /home/rsc/workspace/git/tutorials/core-java-8/target/surefire/surefirebooter3119033343586266928.jar /home/rsc/workspace/git/tutorials/core-java-8/target/surefire 2018-12-29T17-55-30_971-jvmRun2 surefire5414877350063350396tmp surefire_24571033818249870520tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.awaitResultsDone(ForkStarter.java:494)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.runSuitesForkOnceMultiple(ForkStarter.java:368)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:292)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:245)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1149)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:978)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:854)
[ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:954)
[ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
[ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
[ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[ERROR] Caused by: org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/rsc/workspace/git/tutorials/core-java-8 && /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java -jar /home/rsc/workspace/git/tutorials/core-java-8/target/surefire/surefirebooter3119033343586266928.jar /home/rsc/workspace/git/tutorials/core-java-8/target/surefire 2018-12-29T17-55-30_971-jvmRun2 surefire5414877350063350396tmp surefire_24571033818249870520tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:671)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:533)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.access$600(ForkStarter.java:115)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter$1.call(ForkStarter.java:362)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter$1.call(ForkStarter.java:342)
[ERROR] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[ERROR] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[ERROR] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[ERROR] 	at java.lang.Thread.run(Thread.java:748)
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

**Comment by eric-martin** [comment]
@rolfschumacher I'm not able to reproduce your issue. Your error message says there are test failures. Can you tell me which test/s are failing?

**Comment by rolfschumacher** [comment]
[core-java-8.log](https://github.com/eugenp/tutorials/files/2717359/core-java-8.log) is the complete result of the command line
rsc@rolf33:~/workspace/git/tutorials/core-java-8$ mvn clean install -Dgib.enabled=false >/tmp/core-java-8.log 2>/tmp/core-java-8.log
Most notably are the lines
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
telling me, that no test has been run.

In target/surefire-reports I can find a dumpstream holding the lines:
# Created on 2018-12-30T18:32:42.450
Error: Could not find or load main class org.apache.maven.surefire.booter.ForkedBooter

# Created on 2018-12-30T18:32:42.460
Error: Could not find or load main class org.apache.maven.surefire.booter.ForkedBooter

# Created on 2018-12-30T18:32:42.471
Error: Could not find or load main class org.apache.maven.surefire.booter.ForkedBoot

Does this tell you something?

**Comment by rolfschumacher** [comment]
As soon as I wrote the failure comment I get an idea for a solution. I am far from understanding the issue, but I solved it with the help of [Mark Goodrich](https://talk.openmrs.org/t/error-could-not-find-or-load-main-class-org-apache-maven-surefire-booter-forkedbooter/20509). He added maven-surefire-plugin with 

> -Djdk.net.URLClassPath.disableClassPathURLCheck=true

whatever that does.
It did not work with 2.17 version of the plugin as suggested by Mark, but with 3.0.0.-M3.
